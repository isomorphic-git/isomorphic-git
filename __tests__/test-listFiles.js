/* eslint-env node, browser, jasmine */
const { makeFixture } = require('./__helpers__/FixtureFS.js')
const snapshots = require('./__snapshots__/test-listFiles.js.snap')
const registerSnapshots = require('./__helpers__/jasmine-snapshots')
const { listFiles } = require('isomorphic-git')

describe('listFiles', () => {
  beforeAll(() => {
    registerSnapshots(snapshots)
  })
  it('index', async () => {
    // Setup
    let { fs, gitdir } = await makeFixture('test-listFiles')
    // Test
    const files = await listFiles({ fs, gitdir })
    expect(files).toMatchSnapshot()
  })
  it('ref', async () => {
    // Setup
    let { fs, gitdir } = await makeFixture('test-checkout')
    // Test
    const files = await listFiles({ fs, gitdir, ref: 'test-branch' })
    expect(files).toMatchSnapshot()
  })
})
