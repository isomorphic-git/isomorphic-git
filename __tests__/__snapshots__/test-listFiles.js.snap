// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`listFiles index 1`] = `
Array [
  ".babelrc",
  ".editorconfig",
  ".flowconfig",
  ".gitignore",
  ".travis.yml",
  "LICENSE.md",
  "README.md",
  "package-lock.json",
  "package.json",
  "shrinkwrap.yaml",
  "src/commands/checkout.js",
  "src/commands/config.js",
  "src/commands/fetch.js",
  "src/commands/init.js",
  "src/index.js",
  "src/models/GitBlob.js",
  "src/models/GitCommit.js",
  "src/models/GitConfig.js",
  "src/models/GitObject.js",
  "src/models/GitTree.js",
  "src/utils/exists.js",
  "src/utils/mkdirs.js",
  "src/utils/read.js",
  "src/utils/resolveRef.js",
  "src/utils/write.js",
  "test/_helpers.js",
  "test/snapshots/test-resolveRef.js.md",
  "test/snapshots/test-resolveRef.js.snap",
  "test/test-clone.js",
  "test/test-config.js",
  "test/test-init.js",
  "test/test-resolveRef.js",
]
`;

exports[`listFiles ref 1`] = `
Array [
  ".babelrc",
  ".editorconfig",
  ".flowconfig",
  ".gitignore",
  "LICENSE.md",
  "README.md",
  "package.json",
  "shrinkwrap.yaml",
  "src/commands/checkout.js",
  "src/commands/config.js",
  "src/commands/fetch.js",
  "src/commands/init.js",
  "src/index.js",
  "src/models/GitBlob.js",
  "src/models/GitCommit.js",
  "src/models/GitConfig.js",
  "src/models/GitTree.js",
  "src/utils/combinePayloadAndSignature.js",
  "src/utils/commitSha.js",
  "src/utils/exists.js",
  "src/utils/mkdirs.js",
  "src/utils/read.js",
  "src/utils/resolveRef.js",
  "src/utils/unwrapObject.js",
  "src/utils/wrapCommit.js",
  "src/utils/write.js",
  "test/resolveRef.js",
  "test/smoke.js",
  "test/snapshots/resolveRef.js.md",
  "test/snapshots/resolveRef.js.snap",
]
`;
